api/environments/identities/models.py

```diff
@@ -73,19 +74,23 @@ def get_all_feature_states(
         :return: (list) flags for an identity with the correct values based on
             identity / segment priorities
         """
-        segments = self.get_segments(traits=traits, overrides_only=True)  # type: ignore[arg-type]
+        identity = self
+        if from_replica:
+            identity = using_database_replica(Identity.objects).get(id=self.id)
```

Comment:
I think we probably need a comment here to explain this logic?

Reply:
Attempted in 0a98febdad8500d31d30b231da84d4d3437045e8.